token/program-2022/src/state.rs

```diff
@@ -80,6 +103,55 @@ impl Pack for Mint {
     }
 }
 
+/// MintWithTransferFee data.
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, PartialEq)]
+pub struct MintWithTransferFee {
+    /// Base mint data, done as composition
+    pub base_mint: Mint,
+
+    /// ALL NEW DATA MUST START FROM HERE
+    /// account type, always set to "MintWithTransferFee"
+    pub account_type: AccountType,
+    /// Any provided fee account must be owned by this key. If set to `None`, then
+    /// this mint behaves like a normal `Mint`.
+    pub fee_account_owner: COption<Pubkey>,
+    /// Amount of transfer collected as fees, expressed as basis points of the
+    /// transfer amount, ie.  increments of 0.01%
+    pub transfer_fee_basis_points: u16,
+    /// Maximum fee assessed on transfers, expressed as an amount of tokens
+    pub maximum_fee: u64,
```

Comment:
Proposal to epoch-lock the fee before we run into the vote-account commission problem

Reply:
:pray: thank you.  I can't even tell you how disappointed I am in myself for forgetting this.